_posts/2022-04-25-tardigrade-launch.md

```diff
@@ -0,0 +1,85 @@
+* When multiple queries are submitted concurrently they would be able to share resources in a fair way and make steady progress
+
+Trino will do all the hard work behind the scenes of allocating, configuring, and maintaining the queries. Instead of spending time tuning Trino clusters to match your workload requirements or reorganizing your workload to match your Trino cluster capabilities you spend your time doing analytics and delivering business value. And most importantly, your heart won’t skip a beat when you wake up in the morning wondering whether that query landed on time.
+
+## What Have We Done To Test It Out?
+
+Since there's no publically available testing query set for ETL use cases, we handcrafted hundreds of ETL-like queries based on the TPC-H and TPC-DS datasets.
+
+To simulate real world settings, we deployed a Tardigrade cluster of 15 `m5.8xlarge` nodes and repeatedly executed thousands of queries over datasets of different sizes (`10GB` / `1TB` / `10TB`). The queries were executed sequentially as well as with concurrency factors of 5, 10, and 20. Failure recovery capabilities were tested by crashing a random node in a cluster every couple of minutes while streaming a live workload. 
+
+## How Do I Enable Granular Retries?
+
+Tardigrade can only be enabled for an entire cluster.
+
+In general, we recommend splitting your long-running ETL queries and short-running interactive queries into different clusters. This will ensure the long running ETL queries are not impacted by resource utilization spikes created by interactive workload. Please note that any short-running, interactive queries on a Tardigrade cluster may experience higher latencies due to checkpointing mechanisms.
+
+### Recommended hardware
+
+We recommend deploying coordinator and workers on machines with `32` vCPUs, `128GB` memory and `10Gbit/s` network (for example `m5.8xlarge` EC2 instance on AWS).
+
+### Configuration
+
+First you need to create an S3 bucket for the data to be spooled on (currenty only AWS S3 is supported). We recommend configuring a bucket lifecycle rule to automatically expire abandoned objects in an event of a node crash: <img src="/assets/blog/tardigrade-launch/bucket-lifecycle.png"/>
+
+Second you need to configure exchange manager. It can be done by creating the `etc/exchange-manager.properties` configuration file in the Trino working directory on coordinator and on each worker with the following configuration:
+  * `exchange-manager.name=filesystem`
+  * `exchange.base-directories=s3n://<bucket-name>`
+  * `exchange.s3.region=us-east-1`
+  * `exchange.s3.aws-access-key=<access-key>`
+  * `exchange.s3.aws-secret-key=<secret-key>`
+
+And last you need to configure and enable task level retries:
+  * `retry-policy=TASK`
+  * `query.hash-partition-count=50` (more than 50 is currently not supported by the filesystem exchange implementation)
+  * `enable-dynamic-filtering=false` (currently not supported)
```

Comment:
How about distributed sort?

Reply:
We disable it automatically. Actually I wonder if we should do the same for dynamic filtering for now? Then once we support dynamic filtering it can be enabled transparently?

Reply:
I don't know, may better keep it consistent.

Reply:
https://github.com/trinodb/trino/pull/12037